Apps/phcywarectix/ctix_connector.py

```diff
@@ -0,0 +1,336 @@
+                return action_result.set_status(phantom.APP_ERROR, "GET request failed. {}".format(err_msg)), None
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "Unsupported REST method"), None
+
+    def _test_connectivity(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Checking connectivity with Cyware CTIX Platform")
+
+        # REST endpoint for retrieving all Threat Intel sources from CTIX
+        endpoint = "/source/?Expires={}&AccessID={}&Signature={}&page_size=1".format(
+            self._expires, self._access_id, self._generate_signature(self._access_id, self._secret_key, self._expires))
+
+        # Attempt the GET request to CTIX instance and check for successful connection
+        try:
+            status_code, _ = self._make_request("GET", "{}{}".format(self._baseurl, endpoint), self._verify, action_result)
+        except Exception as e:
+            err_msg = self._get_error_message_from_exception(e)
+            self.save_progress(CYWARE_ERR_CONNECTIVITY_TEST)
+            return action_result.set_status(phantom.APP_ERROR, CYWARE_GET_REQ_FAILED.format(err_msg))
+
+        if phantom.is_fail(status_code):
+            self.save_progress(CYWARE_ERR_CONNECTIVITY_TEST)
+            return action_result.get_status()
+
+        if status_code == 200:
+            self.save_progress(CYWARE_SUCC_CONNECTIVITY_TEST)
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "Test Connectivity Failed with status code: {}".format(status_code))
+
+    def _handle_lookup_domain(self, param):
+        action_result = ActionResult(dict(param))
+        self.add_action_result(action_result)
+
+        # check for required input param
+        domain = param["domain"]
+
+        # build full REST endpoint with Auth signature
+        # make GET request to CTIX OpenAPI
+        try:
+            endpoint = "/search/?Expires={}&AccessID={}&Signature={}&domain={}".format(
+                self._expires, self._access_id, self._generate_signature(self._access_id, self._secret_key, self._expires), domain)
+            status_code, response = self._make_request("GET", "{}{}".format(self._baseurl, endpoint), self._verify, action_result)
+        except Exception as e:
+            err_msg = self._get_error_message_from_exception(e)
+            self.save_progress(CYWARE_GET_REQ_FAILED.format(err_msg))
+            return action_result.set_status(phantom.APP_ERROR, "Domain lookup failed. {}".format(err_msg))
+
+        if phantom.is_fail(status_code):
+            return action_result.get_status()
+
+        # check response status_code
+        if status_code == 200:
+            try:
+                if type(response) == list:
+                    response = response[0]
+                if type(response) != dict:

```

Comment:
```suggestion
                if not isinstance(response, dict):

```

same for other uses of `type(...) != ...`